WooCommerce/Classes/POS/TabBar/POSTabCoordinator.swift

```diff
@@ -143,7 +142,36 @@ private extension POSTabCoordinator {
     func presentPOSView(siteID: Int64) {
         Task { @MainActor [weak self] in
             guard let self else { return }
-            let serviceAdaptor = POSServiceLocatorAdaptor()
+
+            // Check local catalog eligibility before initializing infrastructure
+            // Try to use pre-created service from eligibility checker, otherwise create it now
+            let eligibilityService: POSLocalCatalogEligibilityServiceProtocol
+            if let preCreatedService = (eligibilityChecker as? POSTabEligibilityChecker)?.localCatalogEligibilityService {
+                eligibilityService = preCreatedService
+            } else {
+                // Fallback: assume we're POS eligible and create service
+                eligibilityService = await POSLocalCatalogEligibilityService(
```

Comment:
I recognize this is not ideal since it would block the UI with no loading indication after pressing the tab. I tested with the network link conditioner, and the delay is fairly manageable, even with Edge.